src/index.js

```diff
@@ -1,3 +1,148 @@
+  } catch (error) {
+    displayErrorMessage(`Error creating game: ${error.message}`);
+  }
+};
+
+// Function to load gameId from local storage on page load.
+const loadGameIdFromLocalStorage = () => {
+  const storedGameId = localStorage.getItem('gameId');
+  if (storedGameId) {
+    gameId = parseInt(storedGameId, 10); // Parse the stored value as an integer
+  }
+};
+
+// Function to update the leaderboard on the webpage.
+const updateLeaderboard = (leaderboard) => {
+  const boardList = document.querySelector('.board-list');
+  boardList.classList.add('board_list');
+
+  // Clear the current leaderboard.
+  boardList.innerHTML = '';
+
+  // Sort the leaderboard based on scores in descending order.
+  leaderboard.sort((a, b) => b.score - a.score);
+
+  // Iterate through the leaderboard and create list items to display names and scores.
+  leaderboard.forEach((entry, index) => {
+    const listItem = document.createElement('li');
+    listItem.classList.add('scores-list');
+    listItem.textContent = `${index + 1}. ${entry.user}: ${entry.score}`;
+    boardList.appendChild(listItem);
+  });
+};
+
+// Function to fetch scores for the game.
+const fetchScores = async () => {
+  if (!gameId) {
+    return;
+  }
+  try {
+    const response = await fetch(
+      `https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/${gameId}/scores/`,
+    );
+    const data = await response.json();
+    const leaderboard = data.result; // Store the leaderboard data.
+    updateLeaderboard(leaderboard);
+  } catch (error) {
+    displayErrorMessage(`Error fetching scores: ${error.message}`);
+  }
+};
+
+// Function to handle the form submission.
+const handleSubmit = async (event) => {
+  event.preventDefault();
+
+  const nameInput = document.querySelector('.name-input');
+  const scoreInput = document.querySelector('.scores-input');
+
+  const name = nameInput.value;
+  const score = parseInt(scoreInput.value, 10);
+
+  // Validate input
+  if (!name || Number.isNaN(score)) {
+    displayErrorMessage('Ooops! Please enter a valid name and score.');
+    return;
+  }
+
+  // Add the new score to the leaderboard on the server.
+  try {
+    const response = await fetch(
+      `https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/${gameId}/scores/`,
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ user: name, score }),
+      },
+    );
+
+    if (response.ok) {
+      // Refresh the leaderboard after submitting a score.
+      fetchScores();
+    } else {
+      displayErrorMessage(`Error submitting score: ${response.statusText}`);
+    }
+  } catch (error) {
+    displayErrorMessage(`Error submitting score: ${error.message}`);
+  }
+
+  // Clear the input fields.
+  nameInput.value = '';
+  scoreInput.value = '';
+};
+
+// Function to reload the page.
+const reloadPage = () => {
+  window.location.reload();
+};
```

Comment:
- [ ] You've made excellent progress by updating your list upon adding a new item. However, there's a more efficient way to handle refreshing the list. Instead of reloading the entire window when the Refresh button is clicked, please call `fetchScores();` as specified in the documentation. This will trigger a GET API endpoint request and update the scores list without the need for a full window reload.

By implementing this change, you'll ensure that your project functions as intended, with a smoother user experience when refreshing the list. Keep up the great work! 🌟🔄